android/app/src/main/java/com/dreamteam/tourneynizer/activities/TournamentListActivity.java

```diff
@@ -0,0 +1,100 @@
+import android.graphics.BitmapFactory;
+import android.location.Address;
+import android.location.Geocoder;
+import android.support.v7.app.AppCompatActivity;
+import android.os.Bundle;
+import android.view.Gravity;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.AdapterView;
+import android.widget.LinearLayout;
+import android.widget.ListView;
+import android.widget.ProgressBar;
+
+import com.dreamteam.tourneynizer.R;
+import com.dreamteam.tourneynizer.adapters.TournamentListAdapter;
+import com.dreamteam.tourneynizer.data.Tournament;
+import com.dreamteam.tourneynizer.data.TournamentType;
+
+import java.io.IOException;
+import java.sql.Time;
+import java.util.List;
+import java.util.Timer;
+import java.util.TimerTask;
+
+public class TournamentListActivity extends AppCompatActivity {
+
+    public static final String INTENT_TOURNAMENT = "com.dreamteam.tournament.data.Tournament";
+    private TournamentListAdapter listAdapter;
+
+    // will delete these members when back end is exposed please disregard
+    private Geocoder geocoder;
+    private List<Address> a1;
+    private List<Address> a2;
+    private Bitmap logo;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_tournament_list);
+        ListView listView = findViewById(R.id.tournamentList);
+
+        // sets progress bar to fill list view when empty
+        ProgressBar progressBar = new ProgressBar(this);
+        LinearLayout.LayoutParams layoutParams = new LinearLayout.LayoutParams(LinearLayout.LayoutParams.WRAP_CONTENT, LinearLayout.LayoutParams.MATCH_PARENT);
+        layoutParams.gravity = Gravity.CENTER;
+        progressBar.setLayoutParams(layoutParams);
+        progressBar.setIndeterminate(true);
+        ((ViewGroup) listView.getParent()).addView(progressBar);
+        listView.setEmptyView(progressBar);
+
+        listAdapter = new TournamentListAdapter(this);
+        listView.setAdapter(listAdapter);
+        listView.setOnItemClickListener(new AdapterView.OnItemClickListener() {
+            @Override
+            public void onItemClick(AdapterView<?> adapterView, View view, int i, long l) {
+                startActivity(TournamentInfoActivity.class, listAdapter.getItem(i));
+            }
+        });
+        // when available, request tournament info from back end and add Tournament objects to listAdapter
+        // must add to listAdapter on UI thread, if having trouble use runOnUiThread(Runnable)
+        // all lines in this function will be deleted after back end is exposed so you can disregard
+        geocoder = new Geocoder(this);
+        Timer pretendLoad = new Timer();
+        pretendLoad.schedule(new TimerTask() {
+            @Override
+            public void run() {
+                try {
+                    a1 = geocoder.getFromLocationName("5880 W 75th St, Los Angeles, CA 90045", 1);
```

Comment:
Why is the geo API used here? Shouldn't the data needed to specify location come from the server?